File: src/main/java/nl/elstarit/reactive/model/Preference.java
package nl.elstarit.reactive.model;

import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.Data;
import lombok.EqualsAndHashCode;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.List;

@Data
@Document
@EqualsAndHashCode(callSuper=true)
@JsonInclude(JsonInclude.Include.NON_NULL)
public class Preference extends BaseEntity{

    private String userid;
    private List<Topic> topics;
}

File: frontend/src/app/model/preference.ts
import {Topic} from "./topic";

export class Preference {
  id:number;
  userid:string;
  created:Date;
  modified:Date;
  topics: Topic[];

  constructor(id:number, userid:string, created:Date, modified:Date, topics: Topic[]){
    this.id = id;
    this.userid = userid;
    this.created = created;
    this.modified = modified;
    this.topics = topics;
  }
}
